ee/src/main/java/org/jboss/as/ee/component/deployers/EECleanUpProcessor.java

```diff
@@ -46,6 +48,7 @@ public void deploy(final DeploymentPhaseContext phaseContext) throws DeploymentU
 
     @Override
     public void undeploy(final DeploymentUnit context) {
-
+        final Module module = context.getAttachment(org.jboss.as.server.deployment.Attachments.MODULE);
+        FactoryFinderCache.clearClassLoader(module.getClassLoader());
```

Comment:
Sorry, I did not notice that you already mentioned that. I think calling it here is fine however you should remove the call in stopContext, as there is no need to clear it twice.